rpcs3/Emu/RSX/Capture/rsx_capture.cpp

```diff
@@ -236,12 +236,12 @@ namespace rsx
 						auto fifo = vm::ptr<u16>::make(idxAddr);
 						for (u32 i = 0; i < idxCount; ++i)
 						{
-							u16 index = fifo[i];
-							if (is_primitive_restart_enabled && index == (u16)primitive_restart_index)
+							u32 index = (u32)fifo[i];
```

Comment:
I see a lot of these and I think they are unnecessary as u16 cleanly fits into u32. Clutters the code somewhat imo.